File: CineVibe/CineVibe.Services/Database/ProductionCompany.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CineVibe.Services.Database
{
    public class ProductionCompany
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(1000)]
        public string? Description { get; set; }
        
        [MaxLength(100)]
        public string? Country { get; set; }
        
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        // Navigation properties
        public virtual ICollection<MovieProductionCompany> MovieProductionCompanies { get; set; } = new List<MovieProductionCompany>();
    }
}

File: CineVibe/UI/cinevibe_desktop/lib/model/production_company.dart
import 'package:json_annotation/json_annotation.dart';

part 'production_company.g.dart';

@JsonSerializable()
class ProductionCompany {
  final int id;
  final String name;
  final String? description;
  final String? country;
  final bool isActive;
  final DateTime createdAt;
  final int movieCount;

  ProductionCompany({
    required this.id,
    required this.name,
    this.description,
    this.country,
    required this.isActive,
    required this.createdAt,
    required this.movieCount,
  });

  factory ProductionCompany.fromJson(Map<String, dynamic> json) => _$ProductionCompanyFromJson(json);

  Map<String, dynamic> toJson() => _$ProductionCompanyToJson(this);
}
